### ISSUE TITLE
vagrant failing at downloading VirtualBox image for fedora/30-cloud-base

### ISSUE THREAD

**Issue by talamoig** [bpftools/linux-observability-with-bpf#54]
Running `vagrant up` fails since the image for the specific box version (30.20190425.0) cannot be downoaded:
```
==> bpfbook: Box 'fedora/30-cloud-base' could not be found. Attempting to find and install...
    bpfbook: Box Provider: virtualbox
    bpfbook: Box Version: 30.20190425.0
==> bpfbook: Loading metadata for box 'fedora/30-cloud-base'
    bpfbook: URL: https://vagrantcloud.com/fedora/30-cloud-base
==> bpfbook: Adding box 'fedora/30-cloud-base' (v30.20190425.0) for provider: virtualbox
    bpfbook: Downloading: https://vagrantcloud.com/fedora/boxes/30-cloud-base/versions/30.20190425.0/providers/virtualbox.box
    bpfbook: Download redirected to host: download.fedoraproject.org
An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and try
again.

The requested URL returned error: 404 Not Found
```

**Comment by talamoig** [comment]
I also went for fedora/33-cloud-base box but that's shipped with kernel 5.8.15 and already chapter 2 example does not work

**Comment by ish-xyz** [comment]
It's not an error related to vagrant itself, the image is not available. I managed to use a mirror to download mine:
https://ftp-stud.hs-esslingen.de/pub/Mirrors/archive.fedoraproject.org/fedora/linux/releases/30/Cloud/x86_64/images/

You get the full list of fedora mirrors here:

https://admin.fedoraproject.org/mirrormanager/

To solve your problem follow these steps:

1. Add new image:

```
vagrant box add fedora30 https://ftp-stud.hs-esslingen.de/pub/Mirrors/archive.fedoraproject.org/fedora/linux/releases/30/Cloud/x86_64/images/Fedora-Cloud-Base-Vagrant-30-1.2.x86_64.vagrant-virtualbox.box
```

2. Edit the Vagrantfile as follow:

```
# -*- mode: ruby -*-
# vi: set ft=ruby :

VAGRANTFILE_API_VERSION = "2"

$bootstrap=<<SCRIPT
dnf install make glibc-devel.i686 elfutils-libelf-devel wget tar vim tmux jq systemtap-sdt-devel clang bcc strace kernel-devel-5.0.9-301.fc30 -y
SCRIPT

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
	ipv4 = "192.168.33.10"
	config.vm.define "bpfbook" do |fedora|
		fedora.vm.box = "fedora30"
#		fedora.vm.box_version = "30.20190425.0"
		net_index = 1
		fedora.vm.hostname = "bpfbook"
		fedora.vm.provider "virtualbox" do |vb|
			vb.customize ["modifyvm", :id, "--memory", "1024"]
		end
		fedora.vm.provider "libvirt" do |lv|
			lv.memory = 1024
		end
		fedora.vm.network :private_network, ip: "#{ipv4}"
		fedora.vm.provision :shell, inline: $bootstrap, :args => "#{ipv4}"
	end
end
```

3. Run `vagrant up`

(output)
```
==> bpfbook: Importing base box 'fedora30'...
==> bpfbook: Matching MAC address for NAT networking...
==> bpfbook: Setting the name of the VM: linux-observability-with-bpf_bpfbook_1608389287486_2871
==> bpfbook: Clearing any previously set network interfaces...
==> bpfbook: Preparing network interfaces based on configuration...
    bpfbook: Adapter 1: nat
    bpfbook: Adapter 2: hostonly
==> bpfbook: Forwarding ports...
    bpfbook: 22 (guest) => 2222 (host) (adapter 1)
==> bpfbook: Running 'pre-boot' VM customizations...
==> bpfbook: Booting VM...
==> bpfbook: Waiting for machine to boot. This may take a few minutes...
    bpfbook: SSH address: 127.0.0.1:2222
    bpfbook: SSH username: vagrant
    bpfbook: SSH auth method: private key
    bpfbook: 
    bpfbook: Vagrant insecure key detected. Vagrant will automatically replace
    bpfbook: this with a newly generated keypair for better security.
    bpfbook: 
    bpfbook: Inserting generated public key within guest...
    bpfbook: Removing insecure key from the guest if it's present...
    bpfbook: Key inserted! Disconnecting and reconnecting using new SSH key...
==> bpfbook: Machine booted and ready!
==> bpfbook: Checking for guest additions in VM...
    bpfbook: The guest additions on this VM do not match the installed version of
    bpfbook: VirtualBox! In most cases this is fine, but in rare cases it can
    bpfbook: prevent things such as shared folders from working properly. If you see
    bpfbook: shared folder errors, please make sure the guest additions within the
    bpfbook: virtual machine match the version of VirtualBox you have installed on
    bpfbook: your host and reload your VM.
    bpfbook: 
    bpfbook: Guest Additions Version: 6.0.0 r127566
    bpfbook: VirtualBox Version: 6.1
==> bpfbook: Setting hostname...
....
```
